### ISSUE TITLE
401 Unauthorized

### ISSUE THREAD

**Issue by delewis13** [akamai/boomerang#212]
Hello,

Attempting a basic implementation of boomerang. Boomerang is called in index.html and attempts to forward results to a Splunk installation on the same host [collecting results via HEC (HTTP Event Collector)]. Whenever I do this, I run into a 401 Unauthorized error.

I can successfully curl to the server and send events that way, but can't for the life of me get Boomerang to do the same.

I am a noob, so it's likely a very basic mistake. My apologies for that.

Index.html is configured as:

![boomerangindex](https://user-images.githubusercontent.com/20104413/44399050-72cde980-a589-11e8-8c10-31893615c866.PNG)

Curl can be seen to succeed below:

![curlsucceeds](https://user-images.githubusercontent.com/20104413/44399052-73668000-a589-11e8-8caa-942112b70676.PNG)

The website gives the following results:

![websiteresult](https://user-images.githubusercontent.com/20104413/44399057-7497ad00-a589-11e8-8165-e3df362bb085.PNG)

On the splunk side of things, HEC is configured like:

![hecconfig1](https://user-images.githubusercontent.com/20104413/44399054-73668000-a589-11e8-863e-d1a352b96842.PNG)
![hecconfig2](https://user-images.githubusercontent.com/20104413/44399055-73ff1680-a589-11e8-93f1-f2023a7afde7.PNG)

Any help is much appreciated!

Cheers,
Daniel

**Comment by bluesmoon** [comment]
Can you see that error in the network panel in your browser and when you do, get the full headers that it's sending?

**Comment by delewis13** [comment]
Thanks for the suggestion [again, complete noob at web development]!

With what you said in mind I looked through the headers, and noted that XMLWithCredentials was set to false.

Edited boomerang.JS to set xhr.withCredentials = true, updated allowed origins to explicitly allow my IP [did not allow CORS = "*" for whatever reason] and went on my merry way. Now successfully ingesting the events!

That said, leads me on to the question of what else did I screw up that caused me to have to do this manual fiddling with the boomerang.JS. Thinking it has something to do with:

"(1534999779853) {h68303ps}: boomerang.rt: [debug] Called done: load
(1534999779859) {h68303ps}: boomerang.rt: [debug] Got start time: 1534999779272
(1534999779866) {h68303ps}: boomerang.rt: [debug] Setting cookie (timer=undefined)

(1534999779867) {h68303ps}: boomerang: [warn] Saved cookie value doesn't match what we tried to set:
(1534999779868) {h68303ps}: boomerang.rt: [error] cannot set start cookie"
